docs/semantics.md

```diff
@@ -82,14 +87,26 @@ Warning: Using stat caching breaks the consistency guarantees discussed in this
 - Cloud Storage FUSE sends Cloud Storage stat requests for the object named ```foo``` and the object named ```foo/```. The first will hit in the stat cache, but the second will have to go all the way to Cloud Storage to receive a negative result.
 
 The negative result for ```foo/``` will be cached, but that only helps with the second invocation of ```ls -l```.
 
-To alleviate this, Cloud Storage FUSE supports a "type cache" on directory inodes. When ```--type-cache-ttl``` is set, each directory inode will maintain a mapping from the name of its children to whether those children are known to be files or directories or both. When a child is looked up, if the parent's cache says that the child is a file but not a directory, only one Cloud Storage object will need to be stated. Similarly if the child is a directory but not a file.
+To alleviate this, Cloud Storage FUSE supports a "type cache" on directory inodes. When type cache is enabled, each directory inode will maintain a mapping from the name of its children to whether those children are known to be files or directories or both. When a child is looked up, if the parent's cache says that the child is a file but not a directory, only one Cloud Storage object will need to be stated. Similarly if the child is a directory but not a file.
+
+The behavior of type cache is controlled by the following flags/config parameters:
+1. Type-cache capacity: The size of the type cache is **not configurable**. By default the type cache will hold up to 32768 items per directory. If you have folders containing more than 32768 items (folders or files) you may want to increase this, otherwise the caching will not function properly when listing that folder's contents:
+    - ListObjects will return information on the items within the folder. Each item's data is cached
+    - Because there are more objects than cache capacity, the earliest entries will be evicted
+    - The linux kernel then asks for a little more information on each file.
+    - As the earliest cache entries were evicted, this is a fresh List request
+    - This cycle repeats and sends a List request for every page in the folder, as though caching were disabled
+
+2. Type-cache TTL: It controls the duration for which Cloud Storage FUSE allows the kernel to inode type attributes. It can be set in one of the following two ways.
+   1. ```metadata-cache: ttl-secs``` in the config-file. This is set as an integer, which sets the TTL in seconds. If this is -1, TTL is taken as infinite i.e. no-TTL based expirations of entries. If this is 0, that disables the type-cache. If this is <-1, then an error is thrown on mount.
+   2. ```--type-cache-ttl``` commandline flag, which can be set to a value like ```10s``` or ```1.5h```. The default is one minute. This will be deprecated in a future version and is currently only available for backward compatibility. If ```metadata-cache: ttl-secs``` is set, ```--type-cache-ttl``` is ignored.
```

Comment:
can we also add a note somewhere common to both stat and type cache that
(a) "metadata-cache:ttl-secs" is the recommended way to set TTL. and stat-cache-ttl and type-cache-ttl flags will be deprecated in future releases.
(b) minimum of stat-cache-ttl and type-cache-ttl is used as TTL when metadata-cache: ttl-secs is not passed.

Reply:
It's there in the caching section above. It was added in the previous PR. I'll move it down here so make it makes more sense.